netket/logging/tar_log.py

```diff
@@ -0,0 +1,163 @@
+# Copyright 2021 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import dataclasses
+import tarfile
+import time
+from io import BytesIO
+
+import os
+from os import path as _path
+import numpy as np
+import jax
+
+from flax import serialization
+
+
+def save_binary_to_tar(tar_file, byte_data, name):
+    abuf = BytesIO(byte_data)
+
+    # Contruct the info object with the correct length
+    info = tarfile.TarInfo(name=name)
+    info.size = len(abuf.getbuffer())
+
+    # actually save the data to the tar file
+    tar_file.addfile(tarinfo=info, fileobj=abuf)
+
+
+class TarLog:
+    """
+        Tar Logger, serializing the variables of the variational state during a run.
```

Comment:
Since this only logs the variational parameters, maybe it should be called `TarStateLog` or similar?

Reply:
ok